### PR TITLE
feat: foundations typography legend

### PR DESCRIPTION
**Pull request by bashirkarimi**

### CODE REVIEW

File: .changeset/beige-jobs-do.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@swisspost/design-system-documentation': minor
+'@swisspost/design-system-styles': minor
+'@swisspost/design-system-tokens': minor
+---
+
+Added Legend element.
```
Comment (gfellerph) [inline .changeset/beige-jobs-do.md:7]:
```suggestion
Added legend element.
```

File: packages/documentation/src/stories/elements/legend/legend.stories.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import type { StoryObj } from '@storybook/web-components';
+import { html } from 'lit';
+import { MetaExtended } from '@root/types';
+
+const meta: MetaExtended = {
+  id: '61faccd2-de2c-48f0-9a06-c051a56580ef',
+  title: 'Typography/Legend',
```
Comment (gfellerph) [inline packages/documentation/src/stories/elements/legend/legend.stories.ts:7]: This needs some input from the other devs (see https://github.com/swisspost/design-system/issues/3640).

```suggestion
  title: 'Foundations/Typography/Legend',
```

File: packages/styles/src/elements/reset.scss [hunk 3]
```diff
@@ -142,3 +142,16 @@ th {
...
+  padding: 0;
+}
+
+fieldset {
+  border: 0;
+}
+
+legend {
+  width: 100%;
+}
```
Comment (gfellerph) [inline packages/styles/src/elements/reset.scss:157]: This would belong into the element styles.

File: packages/styles/src/elements/legend.scss
Comment (gfellerph) [inline packages/styles/src/elements/legend.scss:1]: Should we name this file `fieldset-legend.scss`?

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/styles/src/elements/reset.scss (hunk 3 above)
Comment (bashirkarimi) [reply packages/styles/src/elements/reset.scss]: The `width` style is moved to element styles.

File: packages/styles/src/elements/legend.scss
Comment (bashirkarimi) [reply packages/styles/src/elements/legend.scss:1]: It is renamed.

File: .changeset/beige-jobs-do.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+'@swisspost/design-system-documentation': minor
+'@swisspost/design-system-styles': minor
+'@swisspost/design-system-tokens': minor
```
[the remark this answers is not in the document]
Comment (bashirkarimi) [reply .changeset/beige-jobs-do.md]: The change log for token package is removed.

File: packages/documentation/src/stories/elements/legend/legend.stories.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  `,
+};
+
+export const Large: Story = {
+  render: () => html`
+    <fieldset>
+      <legend class="legend-large">Large legend</legend>
+    </fieldset>
+  `,
+};
```
[the remark this answers is not in the document]
Comment (bashirkarimi) [reply packages/documentation/src/stories/elements/legend/legend.stories.ts]: The stories are combined and control is added.

File: packages/styles/src/elements/fieldset-legend.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+tokens.$default-map: elements.$post-legend;
+
+legend {
+  font-weight: tokens.get('legend', 'font-weight');
+  line-height: tokens.get('legend', 'line-height');
+  width: 100%;
+}
+
+.legend-large {
```
[the remark this answers is not in the document]
Comment (bashirkarimi) [reply packages/styles/src/elements/fieldset-legend.scss]: The class is nested.

**Review by oliverschuerch — APPROVED** [review]

**Comment by bashirkarimi** [comment]
> Please move the files to documentation/src/stories/foundations/ instead of documentation/src/stories/elements

It is moved.

File: .changeset/beige-jobs-do.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+'@swisspost/design-system-documentation': minor
+'@swisspost/design-system-styles': minor
+---
+
+Added legend element.
```
Comment (gfellerph) [inline .changeset/beige-jobs-do.md:6]:
```suggestion
Updated and tokenized styles for the legend element.
```

**Review by gfellerph — APPROVED** [review]
Neat!

**Pull request merged**